src/wait/cell.rs

```diff
@@ -99,7 +99,10 @@ impl<T: Notify> WaitCell<T> {
         let result = match test_dbg!(self.compare_exchange(State::PARKING, State::WAITING, AcqRel))
         {
             Ok(_) => {
-                let _ = panic::catch_unwind(move || drop(prev_waiter));
+                // XXX(eliza): it's kind of sad we have to use
+                // `AssertUnwindSafe` here, because `std::thread::Thread`
+                // contains a `Condvar` on macOS :(
+                let _ = panic::catch_unwind(panic::AssertUnwindSafe(move || drop(prev_waiter)));
```

Comment:
I think you also need to get rid of this super trait (https://github.com/hawkw/thingbuf/blob/main/src/wait.rs#L57) right?

Reply:
oh, whoops, yeah, you're right